### PR TITLE
feat: Setup apollos devices as their own defined type

### PR DESCRIPTION
**Pull request by vinnyjth**
A lot of churches have been asking about this. This makes apollos dataviews really easy to setup, and will apply to the shovel apps (since we still create personal devices with the shovel app) 
Screen Shot 2021-12-02 at 11 20 13 AM
Screen Shot 2021-12-02 at 11 20 25 AM

### CODE REVIEW

**Comment by redreceipt** [comment]
guessing we won't be doing this on 3.0?

File: packages/apollos-data-connector-rock/src/personal-devices/data-source.js [hunk 1]
```diff
@@ -16,17 +35,23 @@ export default class PersonalDevices extends RockApolloDataSource {
     // if we already have a device, shortcut the function;
     const currentUser = await this.context.dataSources.Auth.getCurrentPerson();
 
+    console.log(existing);
```
Comment (redreceipt) [inline packages/apollos-data-connector-rock/src/personal-devices/data-source.js:38]:
```suggestion
```

**Comment by vinnyjth** [comment]
> guessing we won't be doing this on 3.0?

TBD. We haven't fully ironed out our "Shovel first strategy". There's a chance that this code will move to the shovel as well. (Shoveling from our database into rock)

**Comment by redreceipt** [comment]
so better copy this PR to the cluster too then until we decide what to drop completely.

**Comment by redreceipt** [comment]
if it does make it to 3.0, that adds one more rock requirement. I just don't want to lose track of the feature.

**Comment by vinnyjth** [comment]
> so better copy this PR to the cluster too then until we decide what to drop completely.

Correct!

**Comment by vinnyjth** [comment]
> if it does make it to 3.0, that adds one more rock requirement. I just don't want to lose track of the feature.

Yah. To be clear - we need to port the PersonalDevice functionality one way or the other, this code adds a refinement to that functionality instead of creating an entirely new scope.

**Comment by redreceipt** [comment]
you copy this over to the cluster so we don't forget, I'll approve

**Comment by vinnyjth** [comment]
Here @redreceipt 
https://github.com/ApollosProject/apollos-cluster/pull/76

**Review by redreceipt — APPROVED** [review]

**Pull request merged**